sdk/trace/batch_span_processor.go

```diff
@@ -187,6 +187,11 @@ func (f forceFlushSpan) SpanContext() trace.SpanContext {
 
 // ForceFlush exports all ended spans that have not yet been exported.
 func (bsp *batchSpanProcessor) ForceFlush(ctx context.Context) error {
+	// Interrupt if context is already canceled.
+	if err := ctx.Err(); err != nil {
+		return err
+	}
```

Comment:
TBH I am not sure if this something we should do.

I looked at [`net/http`](https://github.com/golang/go/tree/master/src/net/http) package and I do not see such safeguards.

I looked at the whole standard library and such pattern is almost never used. `ctx.Err()` is almost always used inside `case <-ctx.Done():`. Moreover, take a look at the `httpDo` example from https://go.dev/blog/context.